sigmadsp/hardware/adau1701.py

```diff
@@ -0,0 +1,118 @@
+"""This module provides functionality for controlling SigmaDSP hardware, e.g.
+
+- Deploying programs
+- Changing parameter register contents
+- Reading parameter registers
+"""
+import logging
+from typing import Union
+
+from sigmadsp.hardware.dsp import Dsp
+from sigmadsp.helper.conversion import (
+    bytes_to_int16,
+    bytes_to_int32,
+    float_to_frac_5_23,
+    frac_5_23_to_float,
+    int16_to_bytes,
+    int32_to_bytes,
+)
+
+# A logger for this module
+logger = logging.getLogger(__name__)
+
+
+class Adau1701(Dsp):
+    """A class for controlling functionality of Analog Devices Sigma DSPs, especially ADAU1701 series parts."""
+
+    # Addresses and sizes of important registers
+    CONTROL_REGISTER = 0x081C
+    CONTROL_REGISTER_LENGTH = 2
+
+    # All fixpoint (parameter) registers are four bytes long
+    FIXPOINT_REGISTER_LENGTH = 4
+
+    # Safeload registers (address, data)
+    SAFELOAD_REGISTERS = [
+        (0x0815, 0x810),
+        (0x0816, 0x811),
+        (0x0815, 0x812),
+        (0x0818, 0x813),
+        (0x0819, 0x814),
+    ]
+
+    # Safeload register length
+    SAFELOAD_SA_LENGTH = 2
+    SAFELOAD_SD_LENGTH = 5
+
+    def soft_reset(self):
+        """Soft reset the DSP.
+
+        Not available on ADAU1701
+        """
+        logger.info("Soft-resetting the DSP is not available on ADAU1701")
```

Comment:
It would be nice to inform the frontend of this, because logger messages are not visible there. Otherwise, you could fall back to `hard_reset`.

Reply:
https://github.com/elagil/sigmadsp/blob/main/sigmadsp/hardware/dsp.py#L189-L192

Reply:
True, I forgot. This would make a nice infinite loop.